### PR TITLE
Mavenizing the project

### PR DESCRIPTION
**Pull request by carlschroedl**
@happyjack27, we had [briefly discussed](https://github.com/happyjack27/autoredistrict/issues/1#issuecomment-171478208) using a formal build system. What do you think about this? 
I've got to call it quits for today, but I can write more about why adopting this tool would be helpful to autoredistrict sometime in the first half of this week if you're curious.

### CODE REVIEW

**Comment by happyjack27** [comment]
my main concern is barriers to entry.

i don't want to make it so that someone who doesn't know how to use maven
or install anything or anything like that is unneccessarily burdened.  this
is incredibly simple to build, and i'd like to keep it that way.  i've
found other projects to be a pain in the a$$ to build and i don't want to
become one of those.

this doesn't look like such a barrier, though.  i guess would anyone who
just wants to do a raw build have to do any extra steps or learn anything
new?

**Comment by happyjack27** [comment]
it broke it.  it doesn't build.
you forgot to change the "package ..." at the top of the files.
also now that resources moved that path in the code will have to change too.
and why is there geography now java.geography but ui is now java.org.autoredistrict (and no .ui)?

**Comment by carlschroedl** [comment]
Howdy! I'd like to try to reproduce the problem you're describing. Could
you please tell me what commands you were running and what error messages
you observed? On travis and on my workstation, `mvn clean install` is
working fine to build the app.

**Comment by carlschroedl** [comment]
To debug this now that it has been reverted, try pulling down the maven
branch from my fork:
http://stackoverflow.com/questions/5884784/how-to-pull-remote-branch-from-somebody-elses-repo

**Comment by carlschroedl** [comment]
Also, you might have to re-import the project as a maven project in your
IDE to take advantage of the automated goodness.

**Comment by carlschroedl** [comment]
Looks like eclipse supports maven out of the box now:
http://www.eclipse.org/downloads/packages/eclipse-ide-java-ee-developers/mars1
which is cool because a few years back you still had to install a plugin.
Stock Netbeans definitely supports it too.

**Comment by happyjack27** [comment]
no need to debug.  the bug is already known.  the java package heirarchy
wasn't updated.to reflect the refactoring.  for instance, Applet.jar still
says "package ui;" at the top, even though it's now package
main.whatever.whatever.autoredistrict.

i gather that you didn't test build and launch it before commiting because
that would never be able to link right. unless you were just building an
empty directory and running the old jar.


also the ui classes should be in a ui package, and the geography and
solutions packages should be moved inside the autoredistrict package.

and i didn't look close enough to see if it was done or not, but  the java
build and run from command line would need to be changed to reflect the new
locations of the sources (including the run).

**Comment by carlschroedl** [comment]
> no need to debug. the bug is already known. the java package heirarchy wasn't updated.to reflect the refactoring. for instance, Applet.jar still says "package ui;" at the top, even though it's now package main.whatever.whatever.autoredistrict.

Yep, I forgot to update that. Thanks for pointing it out. I apologize for not acknowledging that in my first post-revert response. It turns out that it doesn't matter that the package declarations and import statements haven't been updated, the app still builds and runs using the updated commands that I specified in the README.md, but I will update them nonetheless. The resource names probably still need to be updated. 

> i gather that you didn't test build and launch it before commiting because that would never be able to link right. unless you were just building an empty directory and running the old jar.

I built and launched successfully multiple times using the instructions I placed in the `README.md` before issuing the pull request. Travis also successfully builds.

> also the ui classes should be in a ui package, and the geography and solutions packages should be moved inside the autoredistrict package.

Can do!

> and i didn't look close enough to see if it was done or not, but the java build and run from command line would need to be changed to reflect the new locations of the sources (including the run).

I updated the `README.md` to reflect the new steps to build and run from the command line. I removed the old `build.sh` script, because we don't need it any more with maven. I removed the reference to the `build.sh script` it in the `.travis.yml` file. We don't need to replace it with anything because Travis supports maven natively.

**Comment by carlschroedl** [comment]
TLDR; I'll work on the package declarations and imports, find and redefine the resource paths, and update this PR or a subsequent one.

Are you having problems getting the project to build and run when you have the maven branch checked out using the commands in the README? If so, I can help ya out there.

File: README.md [hunk 1]
```diff
@@ -15,21 +15,16 @@ The -Xmx4096M -Xms1024M arguments tell java to reserve 1GB of memory, and allow
```
Comment (carlschroedl) [inline README.md]: Here's the goods!

**Comment by happyjack27** [comment]
strange that it builds with the package declarations messed up.  i don't
think that's to spec.  (though i've always though package declarations to
be a bit redundant.)

and even if maven can build it - it should build automatically in a capable
IDE.  personally i have no need for maven or any kind of build scripts like
that 'cause it builds automatically out of the box.  would like to keep it
that way for others, too.  eclipse, at least, can't build without the
package declarations right.

**Comment by happyjack27** [comment]
i haven't tried the commands in the readme.  no need to except if you want
me to verify.  eclipse builds and launches for me.

**Comment by happyjack27** [comment]
maybe it build because it wasn't cleaned(class files deleted) first?
i don't see how it could possibly link when the import statements are
wrong.  so it stands to reason that it didn't link.

**Comment by carlschroedl** [comment]
Yeah, I'm not sure why it works without updating packages, but it's not for lack of cleaning. As I specified in the readme, I'm running a clean each time. I deleted the bin directory where the former build script put its stuff early in the development on the maven branch, so it wasn't reading stale stuff in from there.

**Comment by happyjack27** [comment]
ok that makes me a little uncomfortable.  it's not breaking when it's
supposed to break.

**Comment by happyjack27** [comment]
did you launch it and see that it launched and the gui appeared?  i think
java links at runtime and that's why it didn't break - because it didn't
link.

**Comment by carlschroedl** [comment]
I read ya, I'm surprised it works myself. Yep, I launched the app, and yep, the gui appeared.

**Comment by carlschroedl** [comment]
And no exceptions or such like written to stdout.

**Comment by carlschroedl** [comment]
Between a few distractions at home and recovering from a corrupted eclipse workspace, I haven't made any progress, but I've got to call it a night. Will take a brief look tomorrow.

**Comment by happyjack27** [comment]
Did you delete the .jar, too?

It occurs to me that: 
1.) deleting the class files does not imply deleting the jar
2.) a build program may terminate with exit code zero simply to report that the build program itself had no bugs, regardless of whether or not the program being built has bugs.  After all, failing to build a program that has compile errors is proper and successful behavior for a build program.

That's all I got.   Probably out of ideas at this point.

I want to add in the new competitiveness stats that FairVote wants but I don't want to create a merge conflict for you.

**Comment by carlschroedl** [comment]
Thanks for the suggestions. It looks like I'm going to have to take a break
for a few days to a week, so it might be better to go ahead with those
features and let me deal the merge conflicts later.
